operator/src/main/java/com/github/streamshub/console/dependents/ConsoleSecret.java

```diff
@@ -0,0 +1,218 @@
+
+@ApplicationScoped
+@KubernetesDependent(labelSelector = ConsoleResource.MANAGEMENT_SELECTOR)
+public class ConsoleSecret extends CRUDKubernetesDependentResource<Secret, Console> implements ConsoleResource {
+
+    public static final String NAME = "console-secret";
+    private static final Random RANDOM = new SecureRandom();
+
+    @Inject
+    ObjectMapper objectMapper;
+
+    public ConsoleSecret() {
+        super(Secret.class);
+    }
+
+    @Override
+    public String resourceName() {
+        return NAME;
+    }
+
+    @Override
+    protected Secret desired(Console primary, Context<Console> context) {
+        var nextAuth = context.getSecondaryResource(Secret.class).map(s -> s.getData().get("NEXTAUTH_SECRET"));
+        Map<String, String> data = new LinkedHashMap<>(2);
+
+        var nextAuthSecret = nextAuth.orElseGet(() -> encodeString(base64String(32)));
+        data.put("NEXTAUTH_SECRET", nextAuthSecret);
+
+        var consoleConfig = buildConfig(primary, context);
+
+        try {
+            data.put("console-config.yaml", encodeString(objectMapper.writeValueAsString(consoleConfig)));
+        } catch (JsonProcessingException e) {
+            throw new UncheckedIOException(e);
+        }
+
+        updateDigest(context, "console-digest", data);
+
+        return new SecretBuilder()
+                .withNewMetadata()
+                    .withName(instanceName(primary))
+                    .withNamespace(primary.getMetadata().getNamespace())
+                    .withLabels(commonLabels("console"))
+                .endMetadata()
+                .withData(data)
+                .build();
+    }
+
+    private static String base64String(int length) {
+        ByteArrayOutputStream buffer = new ByteArrayOutputStream();
+        OutputStream out = Base64.getEncoder().wrap(buffer);
+
+        RANDOM.ints().limit(length).forEach(value -> {
+            try {
+                out.write(value);
+            } catch (IOException e) {
+                throw new UncheckedIOException(e);
+            }
+        });
+
+        return new String(buffer.toByteArray()).substring(0, length);
+    }
+
+    private ConsoleConfig buildConfig(Console primary, Context<Console> context) {
+        ConsoleConfig config = new ConsoleConfig();
+
+        for (var kafkaRef : primary.getSpec().getKafkaClusters()) {
+            addConfig(context, config, kafkaRef);
+        }
+
+        return config;
+    }
+
+    private void addConfig(Context<Console> context, ConsoleConfig config, KafkaCluster kafkaRef) {
+        String namespace = kafkaRef.getNamespace();
+        String name = kafkaRef.getName();
+        String listenerName = kafkaRef.getListener();
+
+        KafkaClusterConfig kcConfig = new KafkaClusterConfig();
+        kcConfig.setNamespace(namespace);
+        kcConfig.setName(name);
+        kcConfig.setListener(listenerName);
+        config.getKafka().getClusters().add(kcConfig);
+
+        // TODO: add informer for Kafka CRs
```

Comment:
Does this mean at present if the Kafka CR is changed, e.g. the listener address changes the console will not react to the change? Or is Java operator SDK running a timed reconciliation as well as reacting to resource updates?

Reply:
It currently means it will not react to the change, but the SDK does have a timed reconciliation of 10 hours (!). I'll push an update to lower that to 60 seconds or so. Unfortunately, it doesn't seem to be an interval that can be configured at runtime, so I'll also open an issue if there isn't already one.

Reply:
For reference: https://github.com/quarkiverse/quarkus-operator-sdk/issues/893

Reply:
Added a periodic reconciliation interval of 60s. Let me know if that doesn't seem like a reasonable value.